### PR TITLE
Decrease the number of access control checks in StatementAnalyzer

### PR DESCRIPTION
**Pull request by skrzypo987**

### CODE REVIEW

**Review by kokosing — APPROVED** [review]

**Comment by kokosing** [comment]
Merged, thanks!

File: presto-main/src/main/java/io/prestosql/sql/analyzer/StatementAnalyzer.java [hunk 1]
```diff
@@ -1278,19 +1278,16 @@ private void analyzeFiltersAndMasks(Table table, QualifiedObjectName name, Optio
...
-
             ImmutableMap.Builder<Field, List<ViewExpression>> columnMasks = ImmutableMap.builder();
             for (Field field : fields) {
-                if (field.getName().isPresent() && visibleColumns.contains(field.getName().orElseThrow())) {
+                if (field.getName().isPresent()) {
                     List<ViewExpression> masks = accessControl.getColumnMasks(session.toSecurityContext(), name, field.getName().get(), field.getType());
-                    columnMasks.put(field, masks);
 
-                    masks.forEach(mask -> analyzeColumnMask(session.getIdentity().getUser(), table, name, field, accessControlScope, mask));
+                    if (!masks.isEmpty() && checkCanSelectFromColumn(name, field.getName().orElseThrow())) {
```
Comment (ssheikin) [inline presto-main/src/main/java/io/prestosql/sql/analyzer/StatementAnalyzer.java:1286]: field.getName().orElseThrow() can be extracted to a variable

**Review by ssheikin — COMMENTED** [review]
https://github.com/prestosql/presto/pull/6381

File: presto-main/src/main/java/io/prestosql/sql/analyzer/StatementAnalyzer.java (hunk 1 above)
Comment (weiatwork) [reply presto-main/src/main/java/io/prestosql/sql/analyzer/StatementAnalyzer.java:1286]: @skrzypo987 Just to make sure I understand this change. What if there are no masks for a restricted column? Do we not check that like before? I guess my question is, will this change "leak" the restricted columns without masks.

**Pull request merged**